app/src/main/java/org/sil/gatherwords/SessionActivity.java

```diff
@@ -118,18 +130,121 @@ public void save_settings_fab_pressed(View view) {
         startActivity(i);
     }
 
-    private static class InsertSessionsTask extends AsyncTask<Session, Void, Void> {
+    private static class InsertSessionsTask extends AsyncTask<Session, Void, List<Long>> {
         private SessionDao sDAO;
 
         InsertSessionsTask(AppDatabase db) {
             sDAO = db.sessionDao();
         }
 
         @Override
-        protected Void doInBackground(Session... sessions) {
-            sDAO.insertSession(sessions);
+        protected List<Long> doInBackground(Session... sessions) {
+            return sDAO.insertSession(sessions);
+        }
+
+    }
+
+    /**
+     * Async task for inserting words from the selected word list
+     */
+    private static class InsertWordsTask extends AsyncTask<Void, Double, List<Long>> {
+        private WordDao wordDao;
+        private AsyncTask<Session, Void, List<Long>> task;
+        private final ThreadLocal<SessionActivity> sessionActivity = new ThreadLocal<>();
```

Comment:
What is your logic behind using a `ThreadLocal`?  The activity does not belong to this thread.

Reply:
Lint was warning about a data leak. It was something to do with using part of an activity in a static class

Reply:
When the tasks got merged, I keep getting `null` assigned to the variable. I'm going to ignore the warning and put a TODO tag there